modules/local/juicer_pre.nf

```diff
@@ -0,0 +1,29 @@
+process JUICER_PRE {
+    tag "$meta.id"
+    label 'process_medium'
+
+    input:
+    tuple val(meta), path(binary), path(agp), path(fai)
+
+    output:
+    tuple val(meta), path("*alignments_sorted.txt"), emit: pairs
+    path "versions.yml"           , emit: versions
+    
+    when:
+    task.ext.when == null || task.ext.when
+
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    juicer pre ${binary} ${agp} ${fai} ${args} | LC_ALL=C sort -k2,2d -k6,6d \\
+        --parallel=8 -S50G | awk '\$3>=0 && \$7>=0'> alignments_sorted.txt
```

Comment:
I would change everything between the first pipe and output `LC_ALL ... $7 >= 0` as `args2`, then it is generic enough. Also we don't want to hard code `parallel` and `buffer S` values. For example, you have assigned `process_medium` in first try it only asks for 6 cpus but `parallel` will expect 8. Same with memory, it asks for 36 and you request 50 for buffer. In the second try the requested goes up to 12 and 72, so we need to make it dynamic.